Sources/AEPOptimize/OptimizeError.swift

```diff
@@ -0,0 +1,45 @@
+// Delete this line
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+import Foundation
+
+/// AEPOptimizeError class used to create AEPOptimizeError from error details received from Experience Edge.
+@objc(AEPOptimizeError)
+public class AEPOptimizeError: NSObject {
+    public let type: String?
+    public let status: Int?
+    public let title: String?
+    public let detail: String?
+    public var aepError = AEPError.unexpected
+
```

Comment:
Do you think it might be useful to add a `public shouldRetry: Bool` for cases recoverable errors where we think the customer should retry?
As per this [doc](https://git.corp.adobe.com/pages/experience-edge/konductor/#/api/errors?id=handling-4xx-and-5xx-responses), status codes 429, 502, 503 and 504 can be retried.

Reply:
Yes correct, I also thought of keeping this variable but the reason why we didn't add it for now is because in edge SDK, we see that these are already under Recoverable error codes and it will keep on retrying for these hits. Now if that recoverable error still persist and customer retries it will be redundant as edge is still retrying.

Reply:
Makes sense, thank you for clarifying!